tap-agent/src/agent/sender_allocation.rs

```diff
@@ -466,92 +466,132 @@ impl SenderAllocationState {
                 self.config.tap.rav_request_timestamp_buffer_ms * 1_000_000,
                 Some(self.config.tap.rav_request_receipt_limit),
             )
-            .await
-            .map_err(|e| match e {
-                tap_core::Error::NoValidReceiptsForRAVRequest => anyhow!(
-                    "It looks like there are no valid receipts for the RAV request.\
-                 This may happen if your `rav_request_trigger_value` is too low \
-                 and no receipts were found outside the `rav_request_timestamp_buffer_ms`.\
-                 You can fix this by increasing the `rav_request_trigger_value`."
-                ),
-                _ => e.into(),
-            })?;
-        let valid_receipts: Vec<_> = valid_receipts
-            .into_iter()
-            .map(|r| r.signed_receipt().clone())
-            .collect();
-        let rav_response_time_start = Instant::now();
-        let response: JsonRpcResponse<EIP712SignedMessage<ReceiptAggregateVoucher>> = self
-            .http_client
-            .request(
-                "aggregate_receipts",
-                rpc_params!(
-                    "0.0", // TODO: Set the version in a smarter place.
-                    valid_receipts,
-                    previous_rav
-                ),
-            )
             .await?;
-
-        let rav_response_time = rav_response_time_start.elapsed();
-        RAV_RESPONSE_TIME
-            .with_label_values(&[&self.sender.to_string()])
-            .observe(rav_response_time.as_secs_f64());
-
-        // we only save invalid receipts when we are about to store our rav
-        //
-        // store them before we call remove_obsolete_receipts()
-        if !invalid_receipts.is_empty() {
-            warn!(
-                "Found {} invalid receipts for allocation {} and sender {}.",
-                invalid_receipts.len(),
-                self.allocation_id,
-                self.sender
-            );
-
-            // Save invalid receipts to the database for logs.
-            // TODO: consider doing that in a spawned task?
-            self.store_invalid_receipts(invalid_receipts.as_slice())
+        match (
+            expected_rav,
+            valid_receipts.is_empty(),
+            invalid_receipts.is_empty(),
+        ) {
+            // No valid receipts, but there are invalid
```

Comment:
```suggestion
            // All receipts are invalid
```
More straight to the point